File: Alternativas/ej19.py
# -*- coding: utf-8 -*-

#Escribe un programa que pida un número entero entre uno y doce 
#e imprima el número de días que tiene el mes correspondiente.

#almacenar numeroMes (int)
#si numeroMes es mayor a 1 y menor a 12
#si numeroMes = 2 es Febrero 28 dias y 29 si el año es bisiesto
#si no, si numeroMes es menor que 8,calculamos el módulo de 2
#si es 1 tiene 31 dias, si es 0, 30 o 28 si es febrero
#si el numeroMes es mayor que 7, calculamos el módulo 2
#si es 0 tiene 30 dias, si es 1, 31

print("---#Calcular días del mes#---")
numeroMes = int(input("Introduce el número correspondiente a un mes> "))
if((numeroMes>0)and(numeroMes<13)):
    if(numeroMes==2):
        print("El mes es febrero y tiene 28 días,excepto si el año es bisiesto que tendría 29")
    else:
        if(numeroMes<8):
            if(numeroMes%2==1):
                print("Este mes tiene 31 días")
            else:
                print("Este mes tiene 30 días")
        else:
            if(numeroMes%2==1):
                print("Este mes tiene 30 días")
            else:
                print("Este mes tiene 31 días")

File: Java/Alternativas/ej19.java
import java.util.Scanner; //Importar la clase Scanner

/**
 * Escribe un programa que pida un número entero entre uno y doce 
 * e imprima el número de días que tiene el mes correspondiente.
 * @author Alberto Miguel Martinez Jimenez
 */
/**
 * Algoritmo
 * Almacena el mes del año (int)
 * si mes <1 o mes>12 devuelve error
 * si mes <= 7 y modulo2 de mes != 0
 * el mes tienes 31 dias
 * si mes <= 7 y modulo2 de mes == 0
 * el mes tiene 30
 * si el mes = 2 tiene 28
 * si mes > 7 y modulo2 de mes == 0
 * el mes tienes 31 dias
 * si mes > 7 y modulo2 de mes != 0
 */

public class ej19 {

	public static void main(String[] args) {
		int mes;
		Scanner scanner = new Scanner(System.in);//Intanciar el objeto Scanner
		
		System.out.println("---Dias del mes---");
		System.out.println("Introduce el número del mes a consultar su longitud> ");
		mes = scanner.nextInt();
		
		if((mes<1)||(mes>12)) {
			System.out.println("Dato erroneo. Ese mes o existe");
		}else if(mes<=7) {
			if(mes%2!=0) {
				System.out.println("El mes tiene 31 días");
			}else{
				if(mes==2) {
					System.out.println("El mes tiene 28 días");
				}else{
					System.out.println("El mes tiene 30 días");	
				}
			}
		}
	}

}
